providers/azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/NetworkInterfaceCardApiMockTest.java

```diff
@@ -87,6 +88,28 @@ public void listNetworkInterfaceCards() throws InterruptedException {
       assertEquals(nicList.get(1).properties().ipConfigurations().get(0).properties().privateIPAllocationMethod(), "Static");
    }
 
+   public void listAllNetworkInterfaceCards() throws InterruptedException {
+      server.enqueue(jsonResponse("/listallnetworkinterfaces.json"));
+
+      final NetworkInterfaceCardApi nicApi = api.getNetworkInterfaceCardApi(resourcegroup);
+      List<NetworkInterfaceCard> nicList = nicApi.listAll();
+      String path = String
+            .format("/subscriptions/%s/providers/Microsoft.Network/networkInterfaces?%s", subscriptionid, apiVersion);
+
+      assertSent(server, "GET", path);
+      assertTrue(nicList.size() == 3);
+      assertTrue(nicList.get(0).properties().ipConfigurations().size() > 0);
+      assertEquals(nicList.get(0).properties().ipConfigurations().get(0).properties().privateIPAllocationMethod(),
+            "Dynamic");
+      assertTrue(nicList.get(1).properties().ipConfigurations().size() > 0);
+      assertEquals(nicList.get(1).properties().ipConfigurations().get(0).properties().privateIPAllocationMethod(),
+            "Static");
+      assertTrue(nicList.get(2).properties().ipConfigurations().size() > 0);
+      assertNotEquals(IdReference.extractResourceGroup(nicList.get(2).id()), resourcegroup);
+      assertEquals(nicList.get(2).properties().ipConfigurations().get(0).properties().privateIPAllocationMethod(),
+            "Static");
+   }
```

Comment:
Also add a test that exercises the 404 fallback defined for this new method.

Reply:
Forgot that one, sorry